In [1]:
import netifaces as ni

In [2]:
ni.interfaces()

['{3F2D2CE8-34DD-4F35-BF76-8E1D4E297407}',
 '{26395DF7-30E1-11E7-BBC8-806E6F6E6963}',
 '{8DAD365C-D750-4EB2-81AE-D9774D8206E4}']

__Win10 Response__ 
['{3F2D2CE8-34DD-4F35-BF76-8E1D4E297407}',
 '{26395DF7-30E1-11E7-BBC8-806E6F6E6963}',
 '{8DAD365C-D750-4EB2-81AE-D9774D8206E4}']

In [3]:
ni.ifaddresses(ni.interfaces()[0])

{-1000: [{'addr': '08:62:66:45:7c:fd'}],
 2: [{'addr': '192.168.11.135',
   'broadcast': '192.168.11.255',
   'netmask': '255.255.255.0'}],
 23: [{'addr': 'fe80::41b1:689d:2933:4a4b%5',
   'broadcast': 'fe80::ffff:ffff:ffff:ffff%5',
   'netmask': 'ffff:ffff:ffff:ffff::/64'}]}

__Win10 Response__
{-1000: [{'addr': '08:62:66:45:7c:fd'}],
 2: [{'addr': '192.168.11.135',
   'broadcast': '192.168.11.255',
   'netmask': '255.255.255.0'}],
 23: [{'addr': 'fe80::41b1:689d:2933:4a4b%5',
   'broadcast': 'fe80::ffff:ffff:ffff:ffff%5',
   'netmask': 'ffff:ffff:ffff:ffff::/64'}]}

In [4]:
ni.ifaddresses(ni.interfaces()[1])

{-1000: [{'addr': ''}],
 2: [{'addr': '127.0.0.1',
   'broadcast': '127.255.255.255',
   'netmask': '255.0.0.0'}],
 23: [{'addr': '::1',
   'broadcast': '::1',
   'netmask': 'ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff/128'}]}

__Win10 Response__
{-1000: [{'addr': ''}],
 2: [{'addr': '127.0.0.1',
   'broadcast': '127.255.255.255',
   'netmask': '255.0.0.0'}],
 23: [{'addr': '::1',
   'broadcast': '::1',
   'netmask': 'ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff/128'}]}

In [5]:
ni.ifaddresses(ni.interfaces()[2])

{-1000: [{'addr': '00:00:00:00:00:00:00:e0'}],
 23: [{'addr': '2001:0:5ef5:79fb:242d:366a:718f:1259',
   'broadcast': '2001:0:ffff:ffff:ffff:ffff:ffff:ffff',
   'netmask': 'ffff:ffff::/32'},
  {'addr': 'fe80::242d:366a:718f:1259%9',
   'broadcast': 'fe80::ffff:ffff:ffff:ffff%9',
   'netmask': 'ffff:ffff:ffff:ffff::/64'}]}

__Win10 Response__
{-1000: [{'addr': '00:00:00:00:00:00:00:e0'}],
 23: [{'addr': '2001:0:5ef5:79fb:242d:366a:718f:1259',
   'broadcast': '2001:0:ffff:ffff:ffff:ffff:ffff:ffff',
   'netmask': 'ffff:ffff::/32'},
  {'addr': 'fe80::242d:366a:718f:1259%9',
   'broadcast': 'fe80::ffff:ffff:ffff:ffff%9',
   'netmask': 'ffff:ffff:ffff:ffff::/64'}]}

In [6]:
print(ni.AF_LINK)

-1000


__Win10 Response__
-1000

In [7]:
addrs=ni.ifaddresses(ni.interfaces()[0])
addrs[ni.AF_INET]

[{'addr': '192.168.11.135',
  'broadcast': '192.168.11.255',
  'netmask': '255.255.255.0'}]

__Win10 Response__
[{'addr': '192.168.11.135',
  'broadcast': '192.168.11.255',
  'netmask': '255.255.255.0'}]

In [8]:
addrs=ni.ifaddresses(ni.interfaces()[1])
addrs[ni.AF_INET]

[{'addr': '127.0.0.1', 'broadcast': '127.255.255.255', 'netmask': '255.0.0.0'}]

__Win10 Response__
[{'addr': '127.0.0.1', 'broadcast': '127.255.255.255', 'netmask': '255.0.0.0'}]